### PR TITLE
ci: add trivy IaC scan

### PR DESCRIPTION
**Pull request by xopham**

### CODE REVIEW

File: .github/workflows/cicd.yaml [hunk 1]
```diff
@@ -129,6 +129,37 @@ jobs:
           name: safety-report
           path: safety-report.txt
 
+  trivy-misconfiguration:
```
Comment (Starkteetje) [inline .github/workflows/cicd.yaml:132]: Why `mis`-config?

Comment (Starkteetje) [reply .github/workflows/cicd.yaml:132]: I see, but might not be intuitive. Maybe add a `-scanner`?

File: .github/workflows/cicd.yaml [hunk 2]
```diff
@@ -129,6 +129,37 @@ jobs:
...
+          helm template helm > deployment/deployment.yaml
+      - name: Run Trivy on deployment.yaml
+        uses: aquasecurity/trivy-action@master
+        with:
+          scan-type: 'config'
+          scan-ref: 'deployment'
+          hide-progress: false
+          format: 'table'
+          exit-code: '1'
+          ignore-unfixed: true
```
Comment (Starkteetje) [inline .github/workflows/cicd.yaml:150]: Think for config, there shouldn't be unfixed ones and if there were, I'd probably wanna know

Comment (Starkteetje) [reply .github/workflows/cicd.yaml:150]: Not sure whether same for docker, but might well be the same

File: helm/templates/_helpers.tpl [hunk 3]
```diff
@@ -46,9 +46,9 @@ capabilities:
     - ALL
 privileged: false
 readOnlyRootFilesystem: true
-runAsGroup: 2000
```
Comment (Starkteetje) [inline helm/templates/_helpers.tpl:49]: Why 1000:2000 in Dockerfiles, but not in config?

Comment (xopham) [reply helm/templates/_helpers.tpl:49]: trivy recommends >10000. Could try to adjust the Dockerfile :shrug:

Comment (Starkteetje) [reply helm/templates/_helpers.tpl:49]: k8s sec page does 1k:2k:3k https://kubernetes.io/docs/tasks/configure-pod-container/security-context/
Would be interested in why trivy wants it higher

Comment (Starkteetje) [reply helm/templates/_helpers.tpl:49]: Guess it's collision blabla, but our image isn't even clone to using the uids and guids; don't like changing existing code for "tool said pls change" reasons, but not enough to not approve

**Review by Starkteetje — DISMISSED** [review]

**Review by Starkteetje — APPROVED** [review]

**Pull request merged**